connector/countconnector/config_integration_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+package countconnector
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/connector/connectortest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/otelcol/otelcoltest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector/internal/metadata"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/pdatatest/pmetrictest"
+)
+
+// Ensure basic count connector configs produce expected metrics.
+// See ./testdata/config-*.yaml for specifics.
+// Test cases:
+// 1) `count:` with empty subconfig uses defaults.
+// 2) `count::logs:` with empty subconfig uses defaults.
+// 3) `count::logs::<custom metric>` overrides default.
```

Comment:
I'd prefer these comments be next to the test they're relevant to. My concern with the current format is that it would be easy to miss this comment when changing (adding, modifying, removing) tests in this file, leaving this comment outdated pretty quickly.